### ISSUE TITLE
App crash on opening exsiting wallet.

### ISSUE THREAD

**Issue by Onefox** [m2049r/xmrwallet#436]
This is on a one plus 2 with OxygenOS 3.6.1 android version 6.0.1
Other wallets are working. This wallet also worked before.
Its right after the wallet screen opens and the 0.0000 balance is shown.

The latest app with a pre exsiting wallet is crashing with the following error:

````
10-10 22:55:54.941 8480-9122/? E/AndroidRuntime: FATAL EXCEPTION: WalletService
                                                 Process: com.m2049r.xmrwallet, PID: 8480
                                                 java.lang.ArrayIndexOutOfBoundsException: length=3; index=128
                                                     at com.m2049r.xmrwallet.model.WalletManager.queryWalletDevice(WalletManager.java:186)
                                                     at com.m2049r.xmrwallet.service.WalletService.openWallet(WalletService.java:555)
                                                     at com.m2049r.xmrwallet.service.WalletService.loadWallet(WalletService.java:536)
                                                     at com.m2049r.xmrwallet.service.WalletService.start(WalletService.java:500)
                                                     at com.m2049r.xmrwallet.service.WalletService.access$500(WalletService.java:49)
                                                     at com.m2049r.xmrwallet.service.WalletService$ServiceHandler.handleMessage(WalletService.java:296)
                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                     at android.os.Looper.loop(Looper.java:152)
                                                     at com.m2049r.xmrwallet.service.MoneroHandlerThread.run(MoneroHandlerThread.java:73)
````

**Comment by m2049r** [comment]
Thanks for the report @Onefox. Is there something special about this wallet? Is it a Ledger wallet? What software was it created with & when (i.e. version)?

**Comment by Onefox** [comment]
Not that I am aware of its my daily wallet, so it has a bit more traffic than the others (25-40 tx in total).
It was created with the android app some month ago, its not a Ledger wallet so sadly I don't realy know...
I did also some xmr.to tx on there. I used sub address on the wallet.

**Comment by m2049r** [comment]
@Onefox could you please try #439 and see if/where that crashes. if you can't compile yourself i can build it for you.

**Comment by m2049r** [comment]
@Onefox http://node.monerujo.io/monerujo-1x8x2s_universal.apk

**Comment by m2049r** [comment]
also, are you REALLY SURE you see the balance 0.0000 when this crashes?

**Comment by m2049r** [comment]
fixed in v1.8.3 (through https://github.com/monero-project/monero/pull/4574)

**Comment by Onefox** [comment]
Hi, sry for the late reply.

I couldn't Install your APK. 

But the update to 1.8.4 just now fixed it. thanks!
